execution_engine/src/storage/global_state/scratch.rs

```
@@ -246,7 +265,10 @@ impl CommitProvider for ScratchGlobalState {
                 },
             };
 
-            self.cache.write().unwrap().insert(key, value);
+            self.cache
+                .write()
+                .expect("Cache lock should not be poisoned")
```

We can return an error here instead.